stateproof/db.go

```diff
@@ -121,3 +180,72 @@ func rowsToPendingSigs(rows *sql.Rows) (map[basics.Round][]pendingSig, error) {
 
 	return res, rows.Err()
 }
+
+//#endregion
+
+//#region Builders Operations
+func persistBuilder(tx *sql.Tx, rnd basics.Round, b *builder) error {
+	_, err := tx.Exec(insertBuilderForRound, rnd, protocol.Encode(b))
+	return err
+}
+
+func getBuilder(tx *sql.Tx, rnd basics.Round) (builder, error) {
+	row := tx.QueryRow(selectBuilderForRound, rnd)
+	var rawBuilder []byte
+	err := row.Scan(&rawBuilder)
+	if err != nil {
+		return builder{}, fmt.Errorf("getBuilder: builder for round %d not found in the database: %w", rnd, err)
+	}
+	var bldr builder
+	err = protocol.Decode(rawBuilder, &bldr)
+	if err != nil {
+		return builder{}, fmt.Errorf("getBuilder: getBuilder: builder for round %d failed to decode: %w", rnd, err)
+	}
+
+	// Stored Builder is corrupted...
+	if bldr.Builder == nil {
+		return builder{}, fmt.Errorf("getBuilder: builder for round %d is corrupted", rnd)
+	}
+
+	bldr.Builder.AllocSigs()
```

Comment:
I've addressed most of your points here: https://github.com/algorand/go-algorand/pull/4960

As for the `AllocSigs`, I've opted to add a separate function to fetch the `Message` only instead of using `getBuilder`, thus avoiding the allocation. 
I feel a bit uncomfortable lazily allocating the signatures' slice only when adding, instead of preemptively allocating it at `getBuilder`, since that could potentially lead to nil dereference and if we're not careful and I'd prefer no to put that responsibility on the user of `builder`.